frontend/js/src/playlists/components/DuplicateTrackModal.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+
+type DuplicateTrackModalProps = {
+  message: string;
+  dontAskAgain: boolean;
+  setDontAskAgain: (value: boolean) => void;
+};
+
+export default NiceModal.create((props: DuplicateTrackModalProps) => {
+  const modal = useModal();
+  const { message, dontAskAgain: initialDontAskAgain, setDontAskAgain } = props;
+
+  // Tried to use dontAskAgain directly, but the checkbox state wasnt updating when toggled.
+  const [localDontAskAgain, setLocalDontAskAgain] = React.useState(
+    initialDontAskAgain
+  );
+
+  const confirm = () => {
+    setDontAskAgain(localDontAskAgain);
+    modal.resolve(true);
```

Comment:
This will become:
```suggestion
    modal.resolve(true, localDontAskAgain);
```

Same change for `cancel` below